addons/l10n_de/migrations/1.1/post-migrate_update_amls.py

```diff
@@ -0,0 +1,46 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    # The tax report line 68 has been removed as it does not appear in tax report anymore.
+    # But, it was referenced in the account.sales.report
+    # So, we update amls of this line only, to make this report consistent.
+
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    country = env['res.country'].search([('code', '=', 'DE')], limit=1)
+    tags_68 = env['account.account.tag']._get_tax_tags('68', country.id)
+    tags_60 = env['account.account.tag']._get_tax_tags('60', country.id)
+
+    cr.execute(
+        """
+        UPDATE account_account_tag_account_move_line_rel
+        SET account_account_tag_id = (
+                                      CASE WHEN account_account_tag_id = {tag_minus_68} THEN {tag_minus_60}
+                                           WHEN account_account_tag_id =  {tag_plus_68} THEN {tag_plus_60}
+                                          ELSE account_account_tag_id
+                                      END
+                                      )
+        WHERE account_account_tag_id IN ({tag_minus_68}, {tag_plus_68})
+        """.format(tag_minus_60=tags_60.filtered(lambda tag: tag.display_name[0] == '-').id,
+                   tag_plus_60=tags_60.filtered(lambda tag: tag.display_name[0] == '+').id,
+                   tag_minus_68=tags_68.filtered(lambda tag: tag.display_name[0] == '-').id,
+                   tag_plus_68=tags_68.filtered(lambda tag: tag.display_name[0] == '+').id,
+                   )
+    )
+
+    cr.execute(
+        r"""
+        UPDATE account_move_line
+        SET tax_audit = REGEXP_REPLACE(tax_audit, '(?<=(^|\s))68:', '60:')
+        FROM (
+             SELECT aml.id as aml_id
+             FROM account_move_line aml
+             JOIN account_account_tag_account_move_line_rel aatamlr ON aatamlr.account_move_line_id = aml.id
+             WHERE aatamlr.account_account_tag_id IN ({tag_minus_60}, {tag_plus_60})
+             ) aml
+        WHERE id = aml.aml_id
+    """.format(tag_minus_60=tags_60.filtered(lambda tag: tag.display_name[0] == '-').id,
+               tag_plus_60=tags_60.filtered(lambda tag: tag.display_name[0] == '+').id,
+               )
```

Comment:
1) as the 2 tags ids are there we can directly use it instead of setting both of them.
As psycopg2 cast the list to an array in postgres, we have to cast our list into a tuple to have the desired effect.
2) Not very important here but we shouldn't use stuff as `aatamlr` when we can use something a little bit more explicit
```suggestion
            JOIN account_account_tag_account_move_line_rel aml_tag_rel ON aml_tag_rel.account_move_line_id = aml.id
            WHERE aml_tag_rel.account_account_tag_id IN {tags_60}
        ) aml
        WHERE id = aml.aml_id
        """.format(tags_60=tuple(tags_60.ids))
```